Source/WebView/turbo.js

```diff
@@ -103,6 +103,9 @@
         if (Turbo.navigator.locationWithActionIsSamePage(location, options.action)) {
           Turbo.navigator.view.scrollToAnchorFromLocation(location)
           return
+        } else if (this.currentVisit?.location?.href === location.href) {
+          this.visitLocationWithOptionsAndRestorationIdentifier(location, options, Turbo.navigator.restorationIdentifier)
+          return
```

Comment:
Looks like this also fixes this for Turbo 8, which is great: https://github.com/hotwired/turbo-ios/pull/160

I'd preferably like to see something close to the Android implementation, where we call back to the native session, so we can add proper logging around it, like this: https://github.com/hotwired/turbo-android/pull/292

Reply:
I cherry-picked that commit directly from the PR!